src/graph/graph_apis.cc

```diff
@@ -82,33 +82,42 @@ DGL_REGISTER_GLOBAL("graph_index._CAPI_DGLGraphCreateMutable")
 
 DGL_REGISTER_GLOBAL("graph_index._CAPI_DGLGraphCreate")
 .set_body([] (DGLArgs args, DGLRetValue* rv) {
-    const IdArray src_ids = IdArray::FromDLPack(CreateTmpDLManagedTensor(args[0]));
-    const IdArray dst_ids = IdArray::FromDLPack(CreateTmpDLManagedTensor(args[1]));
-    const IdArray edge_ids = IdArray::FromDLPack(CreateTmpDLManagedTensor(args[2]));
-    const bool multigraph = static_cast<bool>(args[3]);
-    const int64_t num_nodes = static_cast<int64_t>(args[4]);
-    const bool readonly = static_cast<bool>(args[5]);
+    const IdArray src_ids = args[0];
+    const IdArray dst_ids = args[1];
+    const bool multigraph = static_cast<bool>(args[2]);
+    const int64_t num_nodes = static_cast<int64_t>(args[3]);
+    const bool readonly = static_cast<bool>(args[4]);
     GraphHandle ghandle;
-    if (readonly)
-      ghandle = new ImmutableGraph(src_ids, dst_ids, edge_ids, num_nodes, multigraph);
-    else
-      ghandle = new Graph(src_ids, dst_ids, edge_ids, num_nodes, multigraph);
+    if (readonly) {
+      // TODO(minjie): The array copy here is unnecessary and adds extra overhead.
+      //   However, with MXNet backend, the memory would be corrupted if we directly
+      //   save the passed-in ndarrays into DGL's graph object. We hope MXNet team
+      //   could help look into this.
```

Comment:
@zheng-da here is an issue we found for MX backend. The issue occurs when we directly borrow the passed-in ndarrays into CSR graph. This is very efficient because there is no memory copy. However, with MX backend, the program will crash due to memory double-free. We ends up with a sad workaround using memory copy. We cannot have a clean reproducible example. I hope you could connect @ylfdq1118 with who is in charge of DLPack in MX team and he could show you step-by-step on how to reproduce it in DGL.

Reply:
The DLPack in MXNet is implemented by https://github.com/wkcn. He isn't in the MX team.